### PR TITLE
EAMxx: Wire se_ftype=2.

### PR DESCRIPTION
**Pull request by ambrad**
Now se_ftype 0 and 2 work with np4 and pg2 physics grids. Previously only 0
worked.

Also clean up how se_ftype 2 is handled in atmosphere_dynamics.cpp. There is no
need to remap increments, etc., when on the np4 physics grid since the remap is
just copying CGLL data to DGLL slots. The cleanup removes some temporary fields
and operations with these.

In namelist_defaults_scream.xml, se_ftype defaults to 0 for the np4 physics grid
and 2 for the pg2 physics grid. This facilitates testing both cases without
needing testmods.

Later, after our v1 CIME test configurations stabilize, it would be good to use
the 'share' feature in tests.py and test both ftypes with each physics grid.

Non-BFB for pg2 v1 tests. BFB otherwise.

### CODE REVIEW

**Comment by ambrad** [comment]
Following are plots of qv, 1.5 days between rows, for (ftype 0, 2) x (np4-Eul, pg2-SL) showing the four cases are similar:
* https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.ambradl/scream/ft2/ne30v1.ft0.r1/qv.html
* https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.ambradl/scream/ft2/ne30v1.ft2.r1/qv.html
* https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.ambradl/scream/ft2/ne30pg2v1.sl.ft0.r1/qv.html
* https://web.lcrc.anl.gov/public/e3sm/diagnostic_output/ac.ambradl/scream/ft2/ne30pg2v1.sl.ft2.r1/qv.html

**Comment by ambrad** [comment]
Test results are as expected: pass on everything except diff against baselines in the pg2 v1 CIME tests on Mappy.

**Comment by ambrad** [comment]
@bartgol when you get a chance, would you scan the changes and decide if you're OK with the simplifications to the original np4-physics ftype=2 code?

**Comment by bartgol** [comment]
I'm confused. I thought ftype2 needed to remap a phisics grid increment to dyn, not Q itself, in order to maintain bfbness. Otherwise, if physics is not present, dyn (with pg2) would see a non-zero tracers forcing, no?

File: components/scream/src/dynamics/homme/atmosphere_dynamics.cpp [hunk 1]
```diff
@@ -371,21 +368,12 @@ void HommeDynamics::initialize_impl (const RunType run_type)
...
-      // Need a tmp for dQ_phys
-      using namespace ShortFieldTagsNames;
-      create_helper_field("FQ_phys",{COL,CMP,LEV},{ncols,qsize,nlevs},pgn);
-      m_p2d_remapper->register_field(m_helper_fields.at("FQ_phys"),m_helper_fields.at("FQ_dyn"));
-    } else {
-      // Can remap Q directly into FQ, tendency computed in pre_process step
-      m_p2d_remapper->register_field(*get_group_out("Q",pgn).m_bundle,m_helper_fields.at("FQ_dyn"));
-    }
+    //  1) remap Q_pgn->FQ_dyn
+    // Remap Q directly into FQ, tendency computed in pre_process step
```
Comment (bartgol) [inline components/scream/src/dynamics/homme/atmosphere_dynamics.cpp:375]: Where is the tendency computed though?

File: components/scream/src/dynamics/homme/atmosphere_dynamics.cpp [hunk 2]
```diff
@@ -570,89 +558,50 @@ void HommeDynamics::homme_pre_process (const int dt) {
...
-        });
-      }
-      // Hard adjustment of qdp
-      ff.tracers_forcing(dt,n0,n0_qdp,true,params.moisture);
-      break;
-    default:
-      EKAT_ERROR_MSG ("Error! Unexpected/unsupported forcing algorithm.\n"
-                      "  ftype: " + std::to_string(Homme::etoi(ftype)) + "\n");
+  //  ftype=2: nothing
+  if (ftype == ForcingAlg::FORCING_0) {
```
Comment (ambrad) [inline components/scream/src/dynamics/homme/atmosphere_dynamics.cpp:577]: @bartgol this is the tendency computation for ftype=0 for both np4 and pg2 physics.

File: components/scream/src/dynamics/homme/atmosphere_dynamics.cpp (hunk 1 above)
Comment (ambrad) [reply components/scream/src/dynamics/homme/atmosphere_dynamics.cpp:375]: See comment below tagging the spot in the code where the ftype=0 tendency is computed (old code).

File: components/scream/src/dynamics/homme/atmosphere_dynamics.cpp (hunk 2 above)
Comment (bartgol) [reply components/scream/src/dynamics/homme/atmosphere_dynamics.cpp:577]: Oh, ftype2 is the hard adjustment, I forgot.

**Review by bartgol — APPROVED** [review]

**Comment by AaronDonahue** [comment]
Lets go ahead and merge.

**Pull request merged**